sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -178,14 +178,14 @@ object StateStoreMetrics {
 trait StateStoreCustomMetric {
   def name: String
   def desc: String
-  def unit: String
+  def unit: String = ""
 }
 
-case class StateStoreCustomSumMetric(name: String, desc: String, unit: String)
+case class StateStoreCustomSumMetric(name: String, desc: String, override val unit: String)
```

Comment:
We also would like to not break these three classes as well, either providing default value or providing two params constructor which would set `unit` as default value (`""`).

Reply:
Adding a parameter to a case class is neither source compatible nor binary compatible. I think it's okay to ignore unit on the graph since the metric name has this information. We can re-consider to add unit when we hear the voice from the user.

Reply:
OK, removing unit.